client/src/components/Overview.jsx

```diff
@@ -2,27 +2,59 @@ import axios from 'axios';
 import React from "react";
 import { connect } from 'react-redux';
 //import { FaFacebookSquare, FaInstagram, FaPinterest, FaTwitter } from 'react-icons/fa' ;
-//import { BsStar, BsStarHalf, BsStarFill } from 'react-icons/bs' ;
+import { BsStar, BsStarHalf, BsStarFill } from 'react-icons/bs';
 import StyleSelector from "./StyleSelector";
 import AddToCart from './AddtoCart';
 
 
 function setDefaultStyle(styles) {
-
+  //console.log(styles);
   const result = styles.filter((style) => style['default?'] === true);
- 
   return result;
 }
 
-class Overview extends React.Component {
- 
+//Finds and returns average of reviews
+//TODO: Handle edge case for no reviews in accordance with business docs
+function reviewAverage(ratings) {
+  let total = 0;
+  let weightedSum = 0;
+  for (let i = 1; i < 6; i++) {
+    // eslint-disable-next-line radix
+    total += parseInt(ratings[i]);
+    weightedSum += i * ratings[i];
+  }
+  return (weightedSum / total);
+}
 
+function generateStars(average) {
+  const filledStars = Math.trunc(average);
+  const filledRemainder = average % 1;
```

Comment:
Im not sure how trunc works. but if you are trying to get the first decimal on line 31 i think its % 10